requests/OmgEthPrice.js

```diff
@@ -74,7 +74,7 @@ const tally = new Witnet.Tally({
 const request = new Witnet.Request()
   .addSource(binance) // Use source 1
   .addSource(gateio) // Use source 2
-  .addSource(coinyep) // Use source 3
+  // .addSource(coinyep) // Use source 3
```

Comment:
Just checked once again, and yes, apparently so. This I get when dry running (i.e. `npx witnet-toolkit try-data-request --hex <..>` the request including COINYEP:
```
(node:23768) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'nanos_since_epoch' of null
    at C:\Users\guill\github\witnet\witnet-price-feed-examples\node_modules\witnet-requests\dist\witnet\toolkit.js:446:79
    at Array.map (<anonymous>)
    at _callee7$ (C:\Users\guill\github\witnet\witnet-price-feed-examples\node_modules\witnet-requests\dist\witnet\toolkit.js:445:68)
    at tryCatch (C:\Users\guill\github\witnet\witnet-price-feed-examples\node_modules\regenerator-runtime\runtime.js:63:40)
    at Generator.invoke [as _invoke] (C:\Users\guill\github\witnet\witnet-price-feed-examples\node_modules\regenerator-runtime\runtime.js:294:22)
    at Generator.next (C:\Users\guill\github\witnet\witnet-price-feed-examples\node_modules\regenerator-runtime\runtime.js:119:21)
    at asyncGeneratorStep (C:\Users\guill\github\witnet\witnet-price-feed-examples\node_modules\witnet-requests\dist\witnet\toolkit.js:38:103)
    at _next (C:\Users\guill\github\witnet\witnet-price-feed-examples\node_modules\witnet-requests\dist\witnet\toolkit.js:40:194)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:23768) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:23768) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Reply:
Perhaps it would be better to remove completely the COINYEP source from this file, instead of just commenting the `. addSource` line ?

Reply:
Do you have the bytecode around? witnet-requests 0.7.2 should handle this error elegantly and give us some more detail!

Reply:
```npx witnet-toolkit try-data-request --hex 0ae20412550801123968747470733a2f2f6170692e62696e616e63652e636f6d2f6170692f76332f7469636b65722f706963653f73796d626f6c3d4f4d474554481a168418778218646570726963658218571a3b9aca00185b12480801122d68747470733a2f2f646174612e676174656170692e696f2f617069322f312f7469636b65722f6f6d675f6574681a15841877821864646c6173748218571a3b9aca00185b125b0801123f68747470733a2f2f636f696e7965702e636f6d2f6170692f76312f3f66726f6d3d4f4d4726746f3d455448266c616e673d657326666f726d61743d6a736f6e1a168418778218646570726963658218571a3b9aca00185b1291010801125a68747470733a2f2f646174612e6d6573736172692e696f2f6170692f76312f6173736574732f6f6d672f6d6574726963732f6d61726b65742d646174613f6669656c64733d6d61726b65745f646174612f70726963655f6574681a3186187782186664646174618218666b6d61726b65745f646174618218646970726963655f6574688218571a3b9aca00185b124d0801122c68747470733a2f2f6170692e62697466696e65782e636f6d2f76312f7075627469636b65722f6f6d676574681a1b8418778218646a6c6173745f70726963658218571a3b9aca00185b12610801123268747470733a2f2f6170692e6b72616b656e2e636f6d2f302f7075626c69632f5469636b65723f706169723d4f4d474554481a2987187782186666726573756c74821866664f4d4745544882186161618216008218571a3b9aca00185b1a0d0a0908051205fa402000001003220d0a0908051205fa40200000100310c0843d180a20c0843d28463080e497d012```

Reply:
**witnet-toolkit v1.4.3**:
```
╔════════════════════════════════════════════╗
║ Witnet data request local execution report ║
╚╤═══════════════════════════════════════════╝
 │
 │  ┌────────────────────────────────────────────────┐
 ├──┤ Retrieval stage                                │
 │  ├────────────────────────────────────────────────┤
 │  │ Number of retrieved data sources: 6            │
 │  └┬───────────────────────────────────────────────┘
 │   │
 │   ├─[ Source #0 (api.binance.com) ]
 |   │  Method: HTTP-GET
 |   │  Complete URL: https://api.binance.com/api/v3/ticker/price?symbol=OMGETH
 |   │  Number of executed operators: 4
 |   │  Execution time: 0.0318 ms
 |   │  Execution trace:
 │   │    [0] HTTP-GET -> String: "{\"symbol\":\"OMGETH\",\"price\":\"0.00156600\"}"
 │   │    [1] .parseJSONMap() -> Map: {"price":{"RadonString":"0.00156600"},"symbol":{"RadonString":"OMGETH"}}
 │   │    [2] .getFloat(price) -> Float: 0.001566
 │   │    [3] .multiply(1000000000) -> Float: 1566000
 │   │    [4] .round() -> Integer: 1566000
 |   │  Result: Integer: 1566000
 |   │
 │   ├─[ Source #1 (data.gateapi.io) ]
 |   │  Method: HTTP-GET
 |   │  Complete URL: https://data.gateapi.io/api2/1/ticker/omg_eth
 |   │  Number of executed operators: 4
 |   │  Execution time: 0.0639 ms
 |   │  Execution trace:
 │   │    [0] HTTP-GET -> String: "{\"quoteVolume\":\"7007.87\",\"baseVolume\":\"10.75027318\",\"highestBid\":\"0.001567\",\"high24hr\":\"0.001601\",\"last\":\"0.001567\",\"lowestAsk\":\"0.001571\",\"elapsed\":\"7ms\",\"result\":\"true\",\"low24hr\":\"0.001496\",\"percentChange\":\"2.55\"}\n"
 │   │    [1] .parseJSONMap() -> Map: {"baseVolume":{"RadonString":"10.75027318"},"elapsed":{"RadonString":"7ms"},"high24hr":{"RadonString":"0.001601"},"highestBid":{"RadonString":"0.001567"},"last":{"RadonString":"0.001567"},"low24hr":{"RadonString":"0.001496"},"lowestAsk":{"RadonString":"0.001571"},"percentChange":{"RadonString":"2.55"},"quoteVolume":{"RadonString":"7007.87"},"result":{"RadonString":"true"}}
 │   │    [2] .getFloat(last) -> Float: 0.001567
 │   │    [3] .multiply(1000000000) -> Float: 1567000
 │   │    [4] .round() -> Integer: 1567000
 |   │  Result: Integer: 1567000
 |   │
 │   ├─[ Source #2 (coinyep.com) ]
 |   │  Method: HTTP-GET
 |   │  Complete URL: https://coinyep.com/api/v1/?from=OMG&to=ETH&lang=es&format=json
 |   │  Number of executed operators: 1
 |   │  Execution time: unknown
 |   │  Execution trace:
 │   │    [0] HTTP-GET -> Error: "HTTP GET response was an HTTP error code: 400"
 |   │  Result: Error: "HTTP GET response was an HTTP error code: 400"
 |   │
 │   ├─[ Source #3 (data.messari.io) ]
 |   │  Method: HTTP-GET
 |   │  Complete URL: https://data.messari.io/api/v1/assets/omg/metrics/market-data?fields=market_data/price_eth
 |   │  Number of executed operators: 6
 |   │  Execution time: 0.0634 ms
 |   │  Execution trace:
 │   │    [0] HTTP-GET -> String: "{\"status\":{\"elapsed\":2,\"timestamp\":\"2021-12-14T15:45:06.07462869Z\"},\"data\":{\"market_data\":{\"price_eth\":0.0015659513274240058}}}"
 │   │    [1] .parseJSONMap() -> Map: {"data":{"RadonMap":{"market_data":{"RadonMap":{"price_eth":{"RadonFloat":0.0015659513274240058}}}}},"status":{"RadonMap":{"elapsed":{"RadonInteger":"2"},"timestamp":{"RadonString":"2021-12-14T15:45:06.07462869Z"}}}}
 │   │    [2] .getMap(data) -> Map: {"market_data":{"RadonMap":{"price_eth":{"RadonFloat":0.0015659513274240058}}}}
 │   │    [3] .getMap(market_data) -> Map: {"price_eth":{"RadonFloat":0.0015659513274240058}}
 │   │    [4] .getFloat(price_eth) -> Float: 0.0015659513274240058
 │   │    [5] .multiply(1000000000) -> Float: 1565951.3274240058
 │   │    [6] .round() -> Integer: 1565951
 |   │  Result: Integer: 1565951
 |   │
 │   ├─[ Source #4 (api.bitfinex.com) ]
 |   │  Method: HTTP-GET
 |   │  Complete URL: https://api.bitfinex.com/v1/pubticker/omgeth
 |   │  Number of executed operators: 4
 |   │  Execution time: 0.1999 ms
 |   │  Execution trace:
 │   │    [0] HTTP-GET -> String: "{\"mid\":\"0.0015666\",\"bid\":\"0.0015654\",\"ask\":\"0.0015678\",\"last_price\":\"0.0015677\",\"low\":\"0.0014962\",\"high\":\"0.0016006\",\"volume\":\"3154.7553768999999998\",\"timestamp\":\"1639496705.5646253\"}"
 │   │    [1] .parseJSONMap() -> Map: {"ask":{"RadonString":"0.0015678"},"bid":{"RadonString":"0.0015654"},"high":{"RadonString":"0.0016006"},"last_price":{"RadonString":"0.0015677"},"low":{"RadonString":"0.0014962"},"mid":{"RadonString":"0.0015666"},"timestamp":{"RadonString":"1639496705.5646253"},"volume":{"RadonString":"3154.7553768999999998"}}
 │   │    [2] .getFloat(last_price) -> Float: 0.0015677
 │   │    [3] .multiply(1000000000) -> Float: 1567700
 │   │    [4] .round() -> Integer: 1567700
 |   │  Result: Integer: 1567700
 |   │
 │   └─[ Source #5 (api.kraken.com) ]
 |      Method: HTTP-GET
 |      Complete URL: https://api.kraken.com/0/public/Ticker?pair=OMGETH
 |      Number of executed operators: 7
 |      Execution time: 0.2569 ms
 |      Execution trace:
 │        [0] HTTP-GET -> String: "{\"error\":[],\"result\":{\"OMGETH\":{\"a\":[\"0.001571740\",\"164\",\"164.000\"],\"b\":[\"0.001565550\",\"502\",\"502.000\"],\"c\":[\"0.001570790\",\"4.72393861\"],\"v\":[\"757.73858725\",\"2016.66367525\"],\"p\":[\"0.001547649\",\"0.001536413\"],\"t\":[34,52],\"l\":[\"0.001501180\",\"0.001501180\"],\"h\":[\"0.001603220\",\"0.001603220\"],\"o\":\"0.001509500\"}}}"
 │        [1] .parseJSONMap() -> Map: {"error":{"RadonArray":[]},"result":{"RadonMap":{"OMGETH":{"RadonMap":{"a":{"RadonArray":[{"RadonString":"0.001571740"},{"RadonString":"164"},{"RadonString":"164.000"}]},"b":{"RadonArray":[{"RadonString":"0.001565550"},{"RadonString":"502"},{"RadonString":"502.000"}]},"c":{"RadonArray":[{"RadonString":"0.001570790"},{"RadonString":"4.72393861"}]},"h":{"RadonArray":[{"RadonString":"0.001603220"},{"RadonString":"0.001603220"}]},"l":{"RadonArray":[{"RadonString":"0.001501180"},{"RadonString":"0.001501180"}]},"o":{"RadonString":"0.001509500"},"p":{"RadonArray":[{"RadonString":"0.001547649"},{"RadonString":"0.001536413"}]},"t":{"RadonArray":[{"RadonInteger":"34"},{"RadonInteger":"52"}]},"v":{"RadonArray":[{"RadonString":"757.73858725"},{"RadonString":"2016.66367525"}]}}}}}}
 │        [2] .getMap(result) -> Map: {"OMGETH":{"RadonMap":{"a":{"RadonArray":[{"RadonString":"0.001571740"},{"RadonString":"164"},{"RadonString":"164.000"}]},"b":{"RadonArray":[{"RadonString":"0.001565550"},{"RadonString":"502"},{"RadonString":"502.000"}]},"c":{"RadonArray":[{"RadonString":"0.001570790"},{"RadonString":"4.72393861"}]},"h":{"RadonArray":[{"RadonString":"0.001603220"},{"RadonString":"0.001603220"}]},"l":{"RadonArray":[{"RadonString":"0.001501180"},{"RadonString":"0.001501180"}]},"o":{"RadonString":"0.001509500"},"p":{"RadonArray":[{"RadonString":"0.001547649"},{"RadonString":"0.001536413"}]},"t":{"RadonArray":[{"RadonInteger":"34"},{"RadonInteger":"52"}]},"v":{"RadonArray":[{"RadonString":"757.73858725"},{"RadonString":"2016.66367525"}]}}}}
 │        [3] .getMap(OMGETH) -> Map: {"a":{"RadonArray":[{"RadonString":"0.001571740"},{"RadonString":"164"},{"RadonString":"164.000"}]},"b":{"RadonArray":[{"RadonString":"0.001565550"},{"RadonString":"502"},{"RadonString":"502.000"}]},"c":{"RadonArray":[{"RadonString":"0.001570790"},{"RadonString":"4.72393861"}]},"h":{"RadonArray":[{"RadonString":"0.001603220"},{"RadonString":"0.001603220"}]},"l":{"RadonArray":[{"RadonString":"0.001501180"},{"RadonString":"0.001501180"}]},"o":{"RadonString":"0.001509500"},"p":{"RadonArray":[{"RadonString":"0.001547649"},{"RadonString":"0.001536413"}]},"t":{"RadonArray":[{"RadonInteger":"34"},{"RadonInteger":"52"}]},"v":{"RadonArray":[{"RadonString":"757.73858725"},{"RadonString":"2016.66367525"}]}}
 │        [4] .getArray(a) -> Array: [{"RadonString":"0.001571740"},{"RadonString":"164"},{"RadonString":"164.000"}]
 │        [5] .getInteger(0) -> Float: 0.00157174
 │        [6] .multiply(1000000000) -> Float: 1571740
 │        [7] .round() -> Integer: 1571740
 |      Result: Integer: 1571740
 │
 │  ┌────────────────────────────────────────────────┐
 ├──┤ Aggregation stage                              │
 │  ├────────────────────────────────────────────────┤
 │  │ Execution time: 0.0175 ms                      │
 │  │ Result is Error: UnsupportedOpNonHomogeneous { operator: \"RadonFilters::DeviationStandard\" }│
 │  └────────────────────────────────────────────────┘
 │
 │  ┌────────────────────────────────────────────────┐
 └──┤ Tally stage                                    │
    ├────────────────────────────────────────────────┤
    │ Execution time: 0.0162 ms                      │
    │ Result is Error: UnsupportedFilter { array: RadonArray { value: [RadonError(RadonError { inner: UnhandledIntercept { inner: Some(UnsupportedOpNonHomogeneous { operator: \"RadonFilters::DeviationStandard\" }), message: None } })], is_homogeneous: true }, filter: \"RadonFilters::DeviationStandard\" }│
    └────────────────────────────────────────────────┘
```

Reply:
"Error" interim results should indeed be filtered by the `RadonFilters::DeviationStandard` filter, but not provoke `UnsupportedOpNonHomogeneous`

Reply:
The witnet node handles errors in agregation stage here:

https://github.com/witnet/witnet-rust/blob/f5a30a29a56a0044701453f3daf374552487a063/node/src/actors/rad_manager/handlers.rs#L64-L67

So most probably that logic is missing in other radon implementations. `evaluate_tally_precondition_clause` is used to ensure that all the results have the same RadonType, to avoid errors such as the one we see here.

Reply:
This issue has been addressed in https://github.com/witnet/witnet-rust/pull/2139